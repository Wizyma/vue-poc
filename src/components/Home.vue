<template>
  <div v-if="name && age">
    <h1 v-if="name">My name is {{ name }}</h1>
    <h2 v-if="age">And i am {{ age }} years old</h2>
  </div>
</template>

<script lang="ts">
import Vue from "vue";

export default Vue.extend({
  name: "Home",
  props: {
    name: {
      type: String,
      default: null
    },
    age: {
      type: Number,
      default: null
    }
  },
  data: () => {
    // @ts-ignore
    console.log(this);
    return {};
  }
});
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
h3 {
  margin: 40px 0 0;
}
ul {
  list-style-type: none;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 10px;
}
a {
  color: #42b983;
}
</style>
