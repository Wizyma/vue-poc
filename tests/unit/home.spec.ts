import Home from "@/components/Home.vue";
import { itRenderVueComponentCorrectly } from "../test-utils";

const mutations = [
  {
    name: "with all props",
    props: {
      name: "Xin Zhao",
      age: 25
    }
  },
  {
    name: "without props",
    props: null
  }
];

describe(Home.name, () => {
  itRenderVueComponentCorrectly(Home, mutations);
});
