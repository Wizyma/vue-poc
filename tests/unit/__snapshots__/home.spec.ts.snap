// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VueComponent VueComponent with all propsshould render correctly  1`] = `
<div>
  <h1>My name is Xin Zhao</h1>
  <h2>And i am 25 years old</h2>
</div>
`;

exports[`VueComponent VueComponent without propsshould render correctly  1`] = ``;
